pkg/trace/traceutil/trace.go

```diff
@@ -109,24 +109,32 @@ func ChildrenMap(t pb.Trace) map[uint64][]*pb.Span {
 //
 // A span is considered top-level if:
 // - it's a root span
-// - its parent is unknown (other part of the code, distributed trace)
-// - its parent belongs to another service (in that case it's a "local root"
+// - OR its parent is unknown (other part of the code, distributed trace)
+// - OR its parent belongs to another service (in that case it's a "local root"
 //   being the highest ancestor of other spans belonging to this service and
 //   attached to it).
-func ComputeTopLevel(t pb.Trace) {
-	// build a lookup map
-	spanIDToIdx := make(map[uint64]int, len(t))
-	for i, span := range t {
-		spanIDToIdx[span.SpanID] = i
+func ComputeTopLevel(trace pb.Trace) {
+	spanIdToIndex := make(map[uint64]int, len(trace))
+	for i, span := range trace {
+		spanIdToIndex[span.SpanID] = i
 	}
 
-	// iterate on each span and mark them as top-level if relevant
-	for _, span := range t {
-		if span.ParentID != 0 {
-			if parentIdx, ok := spanIDToIdx[span.ParentID]; ok && t[parentIdx].Service == span.Service {
-				continue
-			}
+	for _, span := range trace {
+		isRootSpan := span.ParentID == 0
```

Comment:
I personally don't think code should be documented through the means of variable names. If you want to do that, just do:
```go
if span.ParentID == 0 {
    // root spans are top level
    ...
}
```
And not name variables to document things.

Reply:
fixed